src/test/enhetstester/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/behandlingsresultat/BehandlingsresultatOpphørUtilsTest.kt

```diff
@@ -0,0 +1,555 @@
+import no.nav.familie.ks.sak.kjerne.behandling.steg.behandlingsresultat.BehandlingsresultatOpphørUtils.utledOpphørsdatoForNåværendeBehandlingMedFallback
+import no.nav.familie.ks.sak.kjerne.beregning.domene.AndelTilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.endretutbetaling.domene.Årsak
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import java.math.BigDecimal
+import java.time.YearMonth
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class BehandlingsresultatOpphørUtilsTest {
+    val søker = tilfeldigPerson()
+
+    val for3mndSiden = YearMonth.now().minusMonths(3)
+    val for2mndSiden = YearMonth.now().minusMonths(2)
+    val for1mndSiden = YearMonth.now().minusMonths(1)
+    val om1mnd = YearMonth.now().plusMonths(1)
+    val om4mnd = YearMonth.now().plusMonths(4)
+
+    @BeforeEach
+    fun reset() {
+        clearStaticMockk(YearMonth::class)
+    }
+
+    @AfterAll
+    fun clearMocks() {
+        clearAllMocks()
+    }
+
+    @Test
+    fun `hentOpphørsresultatPåBehandling skal returnere IKKE_OPPHØRT dersom nåværende andeler strekker seg lengre enn dagens dato`() {
+        val barn1Aktør = randomAktør()
+        val barn2Aktør = randomAktør()
+
+        val forrigeAndeler =
+            listOf(
+                lagAndelTilkjentYtelse(
+                    fom = for3mndSiden,
+                    tom = om1mnd,
+                    beløp = 1054,
+                    aktør = barn1Aktør,
+                ),
+                lagAndelTilkjentYtelse(
+                    fom = for3mndSiden,
+                    tom = om1mnd,
+                    beløp = 1054,
+                    aktør = barn2Aktør,
+                ),
+            )
+
+        val nåværendeAndeler =
+            listOf(
+                lagAndelTilkjentYtelse(
+                    fom = for3mndSiden,
+                    tom = om1mnd,
+                    beløp = 1054,
+                    aktør = barn1Aktør,
+                ),
+                lagAndelTilkjentYtelse(
+                    fom = for3mndSiden,
+                    tom = om4mnd,
+                    beløp = 1054,
+                    aktør = barn2Aktør,
+                ),
+            )
+
+        val opphørsresultat =
+            hentOpphørsresultatPåBehandling(
+                nåværendeAndeler = nåværendeAndeler,
+                forrigeAndeler = forrigeAndeler,
+                nåværendeEndretAndeler = emptyList(),
+                forrigeEndretAndeler = emptyList(),
+                nåværendePersonResultaterPåBarn = emptyList(),
+                forrigePersonResultaterPåBarn = emptyList(),
+            )
```

Comment:
Det ser ut som at det ikke finnes noen tester hvor `nåværendePersonResultaterPåBarn` og `forrigePersonResultaterPåBarn` blir satt til noe annet enn `emptyList()`. I og med at de aktivt brukes for å finne opphørsresultat burde vi kanskje ha noen tester hvor de er satt også?

Reply:
Godt poeng 🤩 
La til flere tester på det nå!